sets/src/index.ts

```diff
@@ -0,0 +1,876 @@
+  kyuremw: 'Kyurem-White',
+  landorust: 'Landorus-Therian',
+  lycanrocday: 'Lycanroc',
+  lycanrocnight: 'Lycanroc-Midnight',
+  meloettapirouette: 'Meloetta-P',
+  meowstic: 'Meowstic-M',
+  oricorioelectric: 'Oricorio-Pom-Pom',
+  oricoriofire: 'Oricorio',
+  oricorioghost: 'Oricorio-Sensu',
+  oricoriopsychic: "Oricorio-Pa'u",
+  pumpkabooaverage: 'Pumpkaboo',
+  rotomc: 'Rotom-Mow',
+  rotomf: 'Rotom-Frost',
+  rotomh: 'Rotom-Heat',
+  rotoms: 'Rotom-Fan',
+  rotomw: 'Rotom-Wash',
+  shaymins: 'Shaymin-Sky',
+  thundurust: 'Thundurus-Therian',
+  tornadust: 'Tornadus-Therian',
+  wishiwashisolo: 'Wishiwashi',
+  wormadamg: 'Wormadam-Sandy',
+  wormadams: 'Wormadam-Trash',
+};
+
+async function addThirdPartySets(
+  gen: calc.Generation,
+  setsByFormat: { [format: string]: PokemonSets }
+) {
+  for (const site in THIRD_PARTY_SOURCES) {
+    const source = THIRD_PARTY_SOURCES[site];
+    for (const f in source.files) {
+      const { format, gen: g } = Formats.fromString(f);
+      if (!format || gen !== g) continue;
+      const file = source.files[f];
+      const raw = await request(`${source.url}${file}`);
+      const match = raw.match(/var.*?=.*?({.*})/s);
+      if (!match) {
+        error(`Could not find sets: ${site}/${file}`);
+        continue;
+      }
+      const json = JSON5.parse(match[1]) as PokemonSets;
+      let sets = setsByFormat[format];
+      if (!sets) {
+        sets = {};
+        setsByFormat[format] = sets;
+      }
+      let num = 0;
+      for (let pokemon in json) {
+        pokemon = THIRD_PARTY_ALIASES[toID(pokemon)] || pokemon;
+        if (!calc.SPECIES[gen][pokemon]) {
+          error(`Pokemon ${pokemon} does not exist in generation ${gen}`);
+          continue;
+        }
+        sets[pokemon] = sets[pokemon] || {};
+        for (const name in json[pokemon]) {
+          const set = fixThirdParty(format, pokemon, json[pokemon][name]);
+          if (validPokemonSet(gen, format, set)) {
+            if (filter(format, pokemon, set)) continue;
+            sets[pokemon][cleanName(name)] = set;
+            num++;
+          } else {
+            invalid(gen, format, pokemon, name, set);
+          }
+        }
+      }
+      report(gen, format, num, `${site}/${file}`);
+    }
+  }
+}
+
+function fixThirdParty(format: Format, pokemon: string, set: PokemonSet) {
+  set.ability = fixedAbility(format, pokemon) || set.ability;
+  if (set.ivs) {
+    let iv: keyof StatsTable<number>;
+    for (iv in set.ivs) {
+      set.ivs[iv] = Number(set.ivs[iv]);
+    }
+  }
+  if (set.evs) {
+    let ev: keyof StatsTable<number>;
+    for (ev in set.evs) {
+      set.evs[ev] = Number(set.evs[ev]);
+    }
+  }
+  return set;
+}
+
+const OLD_STATISTICS: { [key: string]: string } = {
+  gen1uu: 'http://www.smogon.com/stats/2017-12/chaos/gen1uu-1500.json',
+  gen2uu: 'http://www.smogon.com/stats/2016-08/chaos/gen2uu-1500.json',
+  gen2nu: 'http://www.smogon.com/stats/2018-11/chaos/gen2nu-1500.json',
```

Comment:
I think this link is wrong? Is 2018-11 intentional when the other stats are mostly from 2016? This is the only place I get different results than what is checked in.

Reply:
I double-checked, and that link is correct. GSC NU was completely retiered in Oct 2018 so 2018-11 is the only time it's had a ladder in its current form. The others use stats from 2016 because their ladders were more active then than they have ever been since.

Can you show me the diff for gen 2 between your script and what's checked in?

Reply:
Sorry for assuming the error was on your end - the issue was that because there are no NU analyses whatsoever on Smogon for GSC my script wasn't even bothering to download statistics for the tier at all and thus not calculating usage based sets based on the statistics. This should now be fixed :)